core/txpool/legacypool/tx_overflowpool.go

```diff
@@ -77,34 +78,52 @@ func NewTxOverflowPoolHeap(estimatedMaxSize uint64) *TxOverflowPool {
 	}
 }
 
-func (tp *TxOverflowPool) Add(tx *types.Transaction) {
+func (tp *TxOverflowPool) Add(tx *types.Transaction) bool {
 	tp.mu.Lock()
 	defer tp.mu.Unlock()
 
 	if _, exists := tp.index[tx.Hash()]; exists {
 		// Transaction already in pool, ignore
-		return
+		return false
+	}
+
+	txSlots := numSlots(tx)
+
+	// If the transaction is too big to ever fit (and the pool isn't empty right now), reject it
+	if (uint64(txSlots) > tp.maxSize) || (uint64(txSlots) == tp.maxSize && tp.totalSize != 0) {
```

Comment:
why need this extra check?
```
(uint64(txSlots) == tp.maxSize && tp.totalSize != 0)
```

Reply:
This condition handles the scenario where the new transaction requires exactly the maximum number of slots (txSlots == maxSize).
If the pool is not empty (tp.totalSize != 0), adding such a transaction would necessitate evicting all existing transactions to make room (according to current code).
So we reject the transaction to prevent complete eviction of current transactions.